src/argilla/server/schemas/v1/datasets.py

```diff
@@ -615,12 +615,42 @@ def metadata_parsed(self) -> List[MetadataParsedQueryParam]:
         return [MetadataParsedQueryParam(q) for q in self.metadata]
 
 
-class TextQuery(BaseModel):
-    text: StringQuery
+class VectorQuery(BaseModel):
+    name: str
+    record_id: Optional[UUID] = None
+    value: Optional[List[float]] = None
+    max_results: conint(gt=0, le=500) = PydanticField(default=50)
```

Comment:
Correct me if I'm wrong but I believe that `VectorQuery` should be free of `max_results`, the query should be limited in the number of results by the `limit` parameter that we are receiving in the endpoint right?

Reply:
Totally. We can use the `limit` parameter to provide the number of results to return and remove the `max_results` attribute